### DISCUSSION TITLE
Logging when deployed as Hybrid mode

### DISCUSSION THREAD

**Discussion by ponsakorn30214** [Kong/kong#8116] [Help] [answered]
Hi,

As I understand the Hybrid mode deployment, the requests are going through data plane(s) and the control plane is used to configure data plane(s).

How should I implement the logging plugin?
since the request will be going through the data plane, do I have to install a logging plugin on each and every node?
or is there a way to enable the logging plugin on the control plane to listen to requests being sent to the data plane?

**Comment by flrgh** [comment, marked as answer]
Hi @ponsakorn30214 :wave: 

> As I understand the Hybrid mode deployment, the requests are going through data plane(s) and the control plane is used to configure data plane(s).

Yep!

> How should I implement the logging plugin?
> since the request will be going through the data plane, do I have to install a logging plugin on each and every node?
> or is there a way to enable the logging plugin on the control plane to listen to requests being sent to the data plane?

There isn't a fully "built-in" solution for coalescing the data plane logs to a single location, but this can definitely be accomplished. The most common path is to use a logging plugin to ship data plane logs to a central log service (e.g. AWS CloudWatch Logs, Loggly, SumoLogic, Splunk, Graylog, etc, this list could go on and on).

Kong comes bundled with several logging plugins that can ship logs over [TCP](https://docs.konghq.com/hub/kong-inc/tcp-log/), [UDP](https://docs.konghq.com/hub/kong-inc/udp-log/), and [HTTP](https://docs.konghq.com/hub/kong-inc/http-log/). If you're using a logging service that supports ingestion over one of those protocols, then there's nothing to install on your data-planes--simply enable and configure the plugin from your control-plane (you can do this globally or even on a per-service basis). Once your data-planes receive the updated configuration from the control-plane, they'll begin sending their logs to your log service.

Of course there are more options than TCP/UDP/HTTP. Check out the [logging section](https://docs.konghq.com/hub/#logging) of the Kong plugin hub for more logging methods.

Hope this helps!

**Reply by ponsakorn30214** [reply]
@flrgh Thanks for the idea!

I've tested by implementing http-log to a Logstash server, works wonderfully!
